Error: node_modules/@types/ws/index.d.ts:328:18 - error TS2315: Type 'Server' is not generic.

The package `web3-providers-ws` uses `"@types/ws": "^8.5.3"`. But a [breaking change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65501) was introduced in @types/ws version 8.5.5. So, when the user adds the last version of web3.js, the version 8.5.5 is installed. And the following error comes: 
```sh
Error: node_modules/@types/ws/index.d.ts:328:18 - error TS2315: Type 'Server' is not generic.

328         server?: HTTPServer<V> | HTTPSServer<V> | undefined;
                     ~~~~~~~~~~~~~


Error: node_modules/@types/ws/index.d.ts:328:34 - error TS2315: Type 'Server' is not generic.

328         server?: HTTPServer<V> | HTTPSServer<V> | undefined;
                                     ~~~~~~~~~~~~~~
```

For more check: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65849 and 
 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65501. 

### Suggested Solution

At the package `web3-providers-ws`, use `"@types/ws": "8.5.3"` instead of `"@types/ws": "^8.5.3"` and move it to `devDependencies`.


### Quick Fix
At the project that uses web3.js, try Using the following at package.json:
```json
"overrides": {
  "web3-providers-ws": {
    "@types/ws": "8.5.4"
  }
}
```

--- a/packages/web3-providers-ws/CHANGELOG.md
+++ b/packages/web3-providers-ws/CHANGELOG.md
@@ -105,3 +105,7 @@ Documentation:
 -   Dependencies updated
 
 ## [Unreleased]
+
+### Fixed
+
+-   Ensure a fixed version for "@types/ws": "8.5.3" (#6309)

--- a/packages/web3-providers-ws/package.json
+++ b/packages/web3-providers-ws/package.json
@@ -45,6 +45,7 @@
 		"@types/express": "^4.17.13",
 		"@types/jest": "^28.1.6",
 		"@types/morgan": "^1.9.3",
+		"@types/ws": "8.5.3",
 		"@typescript-eslint/eslint-plugin": "^5.30.7",
 		"@typescript-eslint/parser": "^5.30.7",
 		"eslint": "^8.20.0",
@@ -61,7 +62,6 @@
 		"typescript": "^4.7.4"
 	},
 	"dependencies": {
-		"@types/ws": "^8.5.3",
 		"isomorphic-ws": "^5.0.0",
 		"web3-errors": "^1.0.2",
 		"web3-types": "^1.0.2",

--- a/packages/web3-providers-ws/package.json
+++ b/packages/web3-providers-ws/package.json
@@ -45,7 +45,6 @@
 		"@types/express": "^4.17.13",
 		"@types/jest": "^28.1.6",
 		"@types/morgan": "^1.9.3",
-		"@types/ws": "8.5.3",
 		"@typescript-eslint/eslint-plugin": "^5.30.7",
 		"@typescript-eslint/parser": "^5.30.7",
 		"eslint": "^8.20.0",
@@ -62,6 +61,7 @@
 		"typescript": "^4.7.4"
 	},
 	"dependencies": {
+		"@types/ws": "8.5.3",
 		"isomorphic-ws": "^5.0.0",
 		"web3-errors": "^1.0.2",
 		"web3-types": "^1.0.2",

